src/ports/postgres/modules/utilities/transform_vec_cols.py_in

```diff
@@ -0,0 +1,495 @@
+ sunny    |  8 | high   | unhealthy
+ sunny    |  9 | high   | good
+ rain     | 10 | medium | good
+ sunny    | 11 | none   | good
+ overcast | 12 | medium | moderate
+ overcast | 13 | medium | moderate
+ rain     | 14 | low    | unhealthy
+(14 rows)
+""".format(schema_madlib=schema_madlib)
+        
+        if not message:
+            return summary_string
+        elif message.lower() in ('usage', 'help', '?'):
+            return usage_string
+        elif message.lower() in ('example', 'examples'):
+            return example_string
+        else:
+            return """
+No such option. Use "SELECT {schema_madlib}.vec2cols()" for help.
+        """.format(schema_madlib=schema_madlib)
+
+class cols2vec:
+    def __init__(self):
+        self.get_cols_helper = vec_cols_helper()
+        self.module_name = self.__class__.__name__
+
+    def validate_args(self, source_table, output_table,
+                      list_of_features, list_of_features_to_exclude, cols_to_output):
+        """
+            Function to validate input parameters
+        """
+        input_tbl_valid(source_table, self.module_name)
+        output_tbl_valid(output_table, self.module_name)
+
+        _assert(list_of_features and list_of_features.strip(), "{0}: List of "
+                    "features cannot be empty".format(self.module_name))
+        if list_of_features.strip() != '*':
+            is_var_valid(source_table, list_of_features)
+        
+        if list_of_features_to_exclude:
+            if list_of_features_to_exclude.strip() == "*":
+                plpy.error("{0}: Cannot exclude all columns from being "
+                    "features".format(self.module_name))
+            elif list_of_features.strip() != '*':
+                plpy.info("{0} NOTICE: will exclude given column(s) even though "
+                    "list of features was not *".format(self.module_name))
+
+        is_var_valid(source_table, list_of_features_to_exclude)
+        is_var_valid(source_table, cols_to_output)
+
+    def get_and_validate_feature_types(self, source_table):
+        """
+            This function validates the types of all of the features_to_nest,
+            disallowing any features of type array. Lets underlying platform
+            decide whether to allow 2 elements of different types be put into
+            an array together (ex. integer and text features, if put together,
+            will yield error by the underlying platform)
+        """
+        all_cols_and_types = get_cols_and_types(source_table)
+        distinct_types = set([col_type[1] for col_type in all_cols_and_types
+            if col_type[0] in self.features_to_nest])
+        for expr_type in distinct_types:
+            _assert(not is_valid_psql_type(expr_type, ANY_ARRAY),
+                "{0}: Feature columns to nest cannot be of type array"
+                .format(self.module_name))
+        if len(distinct_types) != 1 and 'boolean' in distinct_types:
```

Comment:
Please help me understand the need for `len(distinct_types) != 1`

Reply:
We want to cast boolean to text only if it is mixed with other types. Creating a vector of multiple boolean columns works without casting.

Reply:
Makes sense. I would suggest changing it to `len(distinct_types) > 1` and adding what you said above as a comment.